From 22382334da72bfe6c14bc5bc400e828dd4f647ea Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 23 Apr 2014 00:53:37 -0400
Subject: [PATCH] Add LivingEntity.getLoot()

Get the common loot this entity might drop on death.
---
 src/main/java/net/minecraft/server/EntityInsentient.java | 1 +
 .../org/bukkit/craftbukkit/entity/CraftLivingEntity.java | 9 +++++++++
 2 files changed, 10 insertions(+)

diff --git a/src/main/java/net/minecraft/server/EntityInsentient.java b/src/main/java/net/minecraft/server/EntityInsentient.java
index 792fac086..b494da8e9 100644
--- a/src/main/java/net/minecraft/server/EntityInsentient.java
+++ b/src/main/java/net/minecraft/server/EntityInsentient.java
@@ -269,6 +269,7 @@ public abstract class EntityInsentient extends EntityLiving {
         return null;
     }
 
+    @Nullable public Item getCommonLoot() { return getLoot(); } // EMC
     @Nullable
     protected Item getLoot() {
         return null;
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
index 14fb474f7..295984d7a 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
@@ -407,6 +407,15 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
         }
     }
 
+    // EMC start
+    public ItemStack getLoot() {
+        if (getHandle() instanceof EntityInsentient) {
+            final net.minecraft.server.Item loot = ((EntityInsentient) getHandle()).getCommonLoot();
+            return new org.bukkit.inventory.ItemStack(org.bukkit.craftbukkit.util.CraftMagicNumbers.getMaterial(loot));
+        }
+        return null;
+    }
+    // EMC end
     public EntityEquipment getEquipment() {
         return equipment;
     }
-- 
2.17.1

